src/jit/lower.cpp

```diff
@@ -1357,10 +1367,80 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
         // If an extra node is returned, splice it in the right place in the tree.
         if (arg != putArg)
         {
-            ReplaceArgWithPutArgOrCopy(ppArg, putArg);
+            ReplaceArgWithPutArgOrBitcast(ppArg, putArg);
+        }
+    }
+}
+
+#ifdef _TARGET_ARMARCH_
+//------------------------------------------------------------------------
+// LowerFloatArg: Lower the float call argument on the arm platform.
+//
+// Arguments:
+//    arg    - The arg node
+//    info   - call argument info
+//    argNum - argument number
+//
+//
+// Return Value:
+//    Return nullptr, if no transformation was done;
+//    return arg if there was in place transformation;
+//    return a new tree if the root was changed.
+//
+GenTreePtr Lowering::LowerFloatArg(GenTreePtr arg, fgArgTabEntryPtr info, unsigned argNum)
+{
+    var_types type    = arg->TypeGet();
+    var_types intType = (type == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
+
+    if (arg->OperIsFieldList())
+    {
+        assert(arg->isContained());
+        assert(argNum == 0);
+
+        unsigned fieldNum = 0;
+        for (GenTreeFieldList *list = arg->AsFieldList(); list != nullptr; list = list->Rest(), fieldNum++)
+        {
+            GenTreePtr node    = list->Current();
+            GenTreePtr intNode = LowerFloatArg(node, info, fieldNum);
+            if (intNode != nullptr)
+            {
+                ReplaceArgWithPutArgOrBitcast(list->pCurrent(), intNode);
```

Comment:
A FIELD_LIST doesn't always pass its fields in registers. This should check, because we shouldn't generate a BITCAST for a stack-passed field.

Reply:
Yes, you are right, this check is right below, line 1416. If it is stack location, then the function returns nullptr.